analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -156,41 +158,41 @@ static int UseValueInside()
         void NullCoalesceAssignment(string a, string b, string c, Options options)
         {
             a ??= "(empty)";
-            if (a == null)  // Noncompliant
-            {               // Secondary
-                throw new ArgumentNullException(nameof(a)); // never executed
+            if (a == null)                                                  // Noncompliant
+            {        
+                throw new ArgumentNullException(nameof(a));                 // never executed
             }
 
-            b ??= null; // Noncompliant {{Change this expression which always evaluates to 'null'.}}
-            if (b == null)  // OK
+            b ??= null;                                                     // FN: NOP
+            if (b == null)                                                  // OK
             {
-                throw new ArgumentNullException(nameof(b)); // OK
+                throw new ArgumentNullException(nameof(b));                 // OK
             }
 
-            if ((c ??= "(empty)") == null)  // Noncompliant
-            {                               // Secondary
-                throw new ArgumentNullException(nameof(c)); // never executed
+            if ((c ??= "(empty)") == null)                                  // Noncompliant
+            {                        
```

Comment:
```suggestion
            {
```
same for lines 178, 183, 188, 270,